device-connectors/src/testflinger_device_connectors/devices/zapper_iot/__init__.py

```diff
@@ -56,16 +60,44 @@ def _validate_configuration(
         if provision_plan:
 
             try:
-                validate_provision_plan(provision_plan)
-
                 # Make sure the user created at provision time is
                 # the same used during the test phase.
                 provision_plan["config"]["username"] = username
                 provision_plan["config"]["password"] = password
 
+                # If the initial login is using console-conf, we need to
+                # check if the user has provided an ubuntu_sso_email.
+                # If not, we raise an error.
+                # ubuntu_sso_email will be used to set the username for the
+                # initial login.
+                run_stages = provision_plan["run_stage"]
+                for stage in run_stages:
+                    if (
+                        isinstance(stage, dict)
+                        and "initial_login" in stage.keys()
+                        and stage["initial_login"].get("method")
+                        == "console-conf"
+                    ):
+                        if ubuntu_sso_email:
+                            provision_plan["config"][
+                                "username"
+                            ] = ubuntu_sso_email
+                            break
+                        else:
+                            raise ValueError(
+                                "ubuntu_sso_email is required "
+                                "when initial login using console-conf"
+                            )
+
+                validate_provision_plan(provision_plan)
+
                 provisioning_data["custom_provision_plan"] = provision_plan
-            except ValueError as e:
+            except (ValueError, KeyError) as e:
                 raise ProvisioningError from e
+        # If there is an ubuntu_sso_email and no provision plan, we use it
+        # as the username.
+        elif ubuntu_sso_email:
+            provisioning_data["username"] = ubuntu_sso_email
```

Comment:
And again, if you enjoyed my suggestions above, this would not be needed anymore.
```suggestion
```